src/events/guild/member/guildMemberAdd.ts

```diff
@@ -29,9 +31,25 @@ export default class extends Event {
 
 		if (member.pending) return;
 
-		const config = await databaseManager.getConfig(member.guild.id)
+		const mute: Record<string, any> = await getLatestMute(member, member.guild.id);
+		if (mute) {
+			const muteRole: Snowflake = <Snowflake>await databaseManager.getMuteRole(<Snowflake>member.guild?.id);
+			if (!muteRole) return;
+
+			member.roles.add(muteRole);
```

Comment:
```suggestion
			await member.roles.add(muteRole);
```
We could also consider adding a reason string here to go in the audit log.

Reply:
Feature request to add in the future #223, but fixed the await thing